src/tests/class.test.js

```diff
@@ -1 +1,181 @@
-test.skip('skip', () => {});
+import request from 'supertest';
+import app from '../app';
+import Class from '../models/class';
+import classSeed from '../seeds/class';
+
+// eslint-disable-next-line no-underscore-dangle
+const classId = classSeed[0]._id.toString();
+
+const mockClassA = {
+  day: ['Monday', 'Saturday'],
+  hour: '18:00',
+  trainer: '646428fc0b6aa64a90624c05',
+  activity: '646428fc0b6aa64a90624c05',
+  slots: 3,
+};
+
+const mockClassB = {
+  day: ['Monday', 'Saturday'],
+  hour: '19:00',
+  trainer: '646428fc0b6aa64a90624c05',
+  activity: '646428fc0b6aa64a90624c05',
+  slots: 3,
+};
+
+const mockClassC = {
+  day: ['Monday', 'Saturday'],
+  hour: '20:00',
+  trainer: '646428fc0b6aa64a90624c05',
+  activity: '646428fc0b6aa64a90624c05',
+  slots: 3,
+};
+
+const mockClassRepeat = {
+  day: ['Tuesday', 'Monday'],
+  hour: '23:07',
+  trainer: '646a3278743738c22171407b',
+  activity: '646428fc0b6aa64a90624c05',
+  slots: 3,
+};
+
+const mockClassInvalidId = {
+  day: ['Tuesday', 'Monday'],
+  hour: '16:07',
+  trainer: '646428fc0b6aa64a90624c0',
+  activity: '646428fc0b6aa64a90624c05',
+  slots: 3,
+};
+
+beforeAll(async () => {
+  await Class.collection.insertMany(classSeed);
+});
+
+describe('GET /api/classes', () => {
+  test('Should return status 200 and false error.', async () => {
+    const response = await request(app).get('/api/classes').send();
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+  });
+  test('Should return an object containing an array of classes.', async () => {
+    const response = await request(app).get('/api/classes').send();
+    expect(response.body).toBeInstanceOf(Object);
+    expect(response.body.data).toEqual(expect.arrayContaining([]));
+    expect(response.body.error).toBeFalsy();
+  });
+  test('List of classes should not be empty.', async () => {
+    const response = await request(app).get('/api/classes').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+    expect(response.body.error).toBeFalsy();
+  });
+  test('Should return a message indicating "Complete list of classes".', async () => {
+    const response = await request(app).get('/api/classes').send();
+    expect(response.body.message).toBe('Complete list of classes.');
+  });
+  test('Should return status 404 and truth error.', async () => {
+    const response = await request(app).get('/api/classe').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+  test('Should return all properties of classes.', async () => {
+    const response = await request(app).get('/api/classes').send();
+    const classes = response.body.data;
+    classes.forEach((classItem) => {
```

Comment:
you can save some lines if you change the lines 83 to 87 for
`classes.forEach((classItem) => {
   expect(classItem).toHaveProperty('day', 'hour', 'trainer', 'activity', 'slots');
});`

Reply:
you can do some like this:

`const props = ['day', 'hour', 'trainer', 'activity', 'slots'];

props.forEach((prop) => {
  expect(classItem).toHaveProperty(prop);
});`

Reply:
you can do some like this:

` const props = ['day', 'hour', 'trainer', 'activity', 'slots'];

props.forEach((prop) => {
expect(classItem).toHaveProperty(prop);
}); `

the code of mica works but, the "toHaveProperty" only verifies if the first prop ( in this case 'day') is included in classItem, but not verifies the rest of the props. You could do toHaveProperty('day', 'hour', 'sarasa') and it will pass the tests anyways.

Reply:
Thank you guys, for your comments, but I've decided to keep it as it was. Initially, I considered Mica approach, but then, as Ignacio pointed out, I realized that I needed to check that all elements in the list have all the properties. I did try to simplify it, but the one that is already there is the one I understand the most.